src/thunderbots/software/network_input/filter/robot_filter.cpp

```diff
@@ -6,14 +6,47 @@ FilteredRobotData RobotFilter::getFilteredData(
     const std::vector<SSLRobotDetection> &new_robot_data)
 {
     FilteredRobotData filtered_data;
+
+    //std::vector<SSLRobotDetection> this_robot_data;
+
+    int data_num =0;
+
     filtered_data.id               = 0;
     filtered_data.position         = Point();
     filtered_data.velocity         = Vector();
     filtered_data.orientation      = Angle::zero();
     filtered_data.angular_velocity = AngularVelocity::zero();
     filtered_data.timestamp        = Timestamp::fromSeconds(0);
 
-    return filtered_data;
+    for(const SSLRobotDetection &robot_data: new_robot_data)
+    {
+        //add all data points up and then average it
+        if(robot_data.id==this->getRobotId())
+        {
+
+            filtered_data.position         += robot_data.position;
+            filtered_data.orientation      += robot_data.orientation;
+            filtered_data.timestamp =  robot_data.timestamp;
+            data_num++;
+        }
+    }
+
+
+
+    //if no new data and timestamp difference is long enough, return filtered data as a indication that
+    //the robot is off the field? if no new data and timestamp difference is small enough, then return current robot position
```

Comment:
Although we can usually assume if this function is being called, data will be provided, I think the most robust solution to this is to return an `optional`

Reply:
std::optional sparkling_heart